operators/src/source/gdal_source.rs

```diff
@@ -0,0 +1,456 @@
+
+#[derive(Eq, PartialEq, Debug, Clone, Copy, Serialize, Deserialize)]
+pub struct Tick {
+    pub year: i32,
+    pub month: u32,
+    pub day: u32,
+    pub hour: u32,
+    pub minute: u32,
+    pub second: u32,
+}
+
+impl Tick {
+    pub fn new(year: i32, month: u32, day: u32, hour: u32, minute: u32, second: u32) -> Self { Self { year, month, day, hour, minute, second } }
+
+    pub fn snap_date<T: Datelike>(&self, date: &T) -> NaiveDate {
+        let y = date.year().div_floor(&self.year) * self.year;
+        let m = date.month().div_floor(&self.month) * self.month;
+        let d = date.day().div_floor(&self.day) * self.day;
+        NaiveDate::from_ymd(y,m,d)
+    }
+
+    pub fn snap_time<T: Timelike>(&self, time: &T) -> NaiveTime {
+        let h = time.hour().div_floor(&self.hour) * self.hour;
+        let m = time.minute().div_floor(&self.minute) * self.minute;
+        let s = time.second().div_floor(&self.second) * self.second;
+        NaiveTime::from_hms(h,m,s)
+    }
+
+    pub fn snap_datetime<T: Datelike + Timelike>(&self, datetime: &T) -> NaiveDateTime {
+        NaiveDateTime::new(self.snap_date(datetime), self.snap_time(datetime))
+    }
+}
+
+/// Parameters for the GDAL Source Operator
+///
+/// # Examples
+///
+/// ```rust
+/// use serde_json::{Result, Value};
+/// use geoengine_operators::Operator;
+/// use geoengine_operators::operators::{GdalSourceParameters, NoSources, RasterSources};
+///
+/// let json_string = r#"
+///     {
+///         "type": "gdal_source",
+///         "params": {
+///                     "base_path": "base_path",
+///                     "file_name_with_time_placeholder": "file_name_with_time_placeholder",
+///                     "time_format": "file_name",
+///                     "channel": 3
+///         }
+///     }"#;
+///
+/// let operator: Operator = serde_json::from_str(json_string).unwrap();
+///
+/// assert_eq!(operator, Operator::GdalSource {
+///     params: GdalSourceParameters {
+///                     base_path: "base_path".into(),
+///                     file_name_with_time_placeholder: "file_name_with_time_placeholder".into(),
+///                     time_format: "file_name".into(),
+///                     tick: None,
+///                     channel: Some(3),
+///     },
+///     sources: Default::default(),
+/// });
+/// ```
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct GdalSourceParameters {
+    pub base_path: PathBuf,
+    pub file_name_with_time_placeholder: String,
+    pub time_format: String,
+    pub tick: Option<Tick>,
+    pub channel: Option<u32>,
+}
+
+/// A provider of tile (size) information for a raster/grid
+pub struct GdalSourceTileGridProvider {
+    pub global_pixel_size: Dim<[Ix; 2]>,
+    pub tile_pixel_size: Dim<[Ix; 2]>,
+    // pub grid_tiles : Vec<Dim<[Ix; 2]>>,
+}
+
+impl GdalSourceTileGridProvider {
+
+    /// generates a vec with `TileInformation` for each tile
+    fn tile_informations(&self) -> Vec<TileInformation> {
```

Comment:
Is this necessary, i.e., can't we compute the information for tiles with some metadata?

Reply:
Please elaborate. Some rasters have tile information. But in generally we will have to specify them.

Reply:
Also i tried to use an iterator but this is not the easiest task. Therefore, i used this solution until the design is more mature.

Reply:
The question is if we have to store the information in an array or if we can compute them if we need them? I was thinking in terms of memory consumption and precomputations.

Reply:
I guess there is a way to do this. However, for an evolving source this serves very well while developing